desc/compute/utils.py

```diff
@@ -91,6 +95,35 @@ def compute(parameterization, names, params, transforms, profiles, data=None, **
         data=data,
         **kwargs,
     )
+
+    # convert data from default 'rpz' basis to 'xyz' basis, if requested by the user
+    parameterization = _parse_parameterization(parameterization)
+    for name in data.keys():
+        if (
+            data_index[parameterization][name]["dim"] == 3  # vector quantity
+            and kwargs.get("basis", "rpz").lower() == "xyz"  # xyz basis requested
+        ):
+            from .geom_utils import rpz2xyz, rpz2xyz_vec
+
+            if name == "x":
+                data[name] = rpz2xyz(data[name])
+            else:
+                if "phi" in data:
+                    data[name] = rpz2xyz_vec(data[name], phi=data["phi"])
+                else:
+                    # compute phi if it's not already in data. Adding phi
+                    # to data at the beginning cause matrix dimension problems
+                    # TODO: Maybe look into that later (why error?)
```

Comment:
what error were you getting? was it just a test failing or an actual error?

Reply:
It was a matrix dimension error. I didn't go too deep into this but when I append phi directly to names before recursive _compute function, dimension of phi was not proper for usage in rpz2xyz_vec function. I felt like appending it at the beginning cause it not to use given grid (that seemed the only reason for dimension difference), or something like that

Reply:
can you post the error? I would think if you just do 
```
if basis == "xyz" and "phi" not in names:
    names += ["phi"]
```
should work fine?

Reply:
I am away from my laptop now, but I can try to send it around dev meeting

Reply:
I'm not certain this will always work, because the `params` and `transforms` were already set for the original list of names and might not contain the info needed to compute `"phi"`

Reply:
`"phi"` (currently, this will change with the general angle stuff) only needs `grid` to be included in `transforms` which I think is almost always the case for basically any compute quantity